### DISCUSSION TITLE
Zod validation for .env throws errors in testing sidebar

### DISCUSSION THREAD

**Discussion by jacknevitt** [vitest-dev/vscode#407] [Q&A]
In my typescript project, I validate the env variables with Zod like this:
```ts
// env.server.ts
import { z } from "zod";

const serverEnvSchema = z.object({
  API_URL: z.string(),
});

export const env = serverEnvSchema.parse(process.env);
```
This is validated at build time and the variables can be used reliably across the app. The `env` is then imported like so:
```ts
// inviteUser.server.ts
import { env } from "~/config/env.server";

export async function inviteUser(email: string) {
  const result = await fetch(
    env.API_URL + "/invite",
    { method: "POST", body: JSON.stringify({ email }) },
  );
  return result;
}
```
And then the test file like so:
```ts
import { inviteUser } from "./inviteUser.server";

describe("inviteUser", () => {
  it("calls the api with expected values", async () => {
    const fetchSpy = vi.spyOn(global, "fetch");

    // mocked with MSW
    await inviteUser("test@email.com");

    expect(fetchSpy).toBeCalledWith(
      expect.stringContaining("/invite"),
      expect.objectContaining({
        method: "POST",
        body: JSON.stringify({ email: "test@email.com"}),
      })
    );
  });
});
```
However, when exploring the vitest tests in the vscode testing panel, there are zod errors shown and the tests fail when run from the sidebar. The tests pass with `npm run test`.

![Screenshot 2024-06-04 at 10 58 44](https://github.com/vitest-dev/vscode/assets/19165624/3f7b6d75-1969-44db-b7c4-16b32a075c59)

Is there a way to set .env variables for the extension? I have a `__mocks__/env.server.ts` file, but mocking the file in the test doesn't help. 

Any suggestions?

**Comment by sheremet-va** [comment]
You can set env in the Vitest config file, or via `vitest.nodeEnv` option in vscode settings.

**Reply by jacknevitt** [reply]
Thanks, I'll try that when it happens again. Strangely, it's working now I've changed branches to work on something else and came back to it.
